ksqldb-common/src/main/java/io/confluent/ksql/serde/WindowInfo.java

```diff
@@ -73,20 +85,35 @@ public boolean equals(final Object o) {
       return false;
     }
     final WindowInfo that = (WindowInfo) o;
+
+    final Optional<OutputRefinement> thisEmit = emitStrategy.isPresent()
```

Comment:
@vcrfxia -- I hope this make sense? The idea is to make `empty` and `changes` equal (also cf `hashCode` below)

Reply:
Makes sense but I have a code style question -- if empty is always the same as EMIT CHANGES, then can be change the constructor to store `OutputRefinement` rather than `Optional<OutputRefinement>` and use `emitStrategy.orElse(OutputRefinement.CHANGES)` in the constructor, in order to simplify these methods here?

Reply:
I did consider it, but was not sure -- as you bring it up, too, seems to be the better choice. Let me update it.